cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/opencv-control.tsx

```diff
@@ -191,35 +209,45 @@ class OpenCVControlComponent extends React.PureComponent<Props & DispatchToProps
             shapesUpdated, isDone, threshold, shapes,
         } = (e as CustomEvent).detail;
         const pressedPoints = convertShapesForInteractor(shapes, 0).flat();
-
         try {
             if (shapesUpdated) {
-                this.latestPoints = await this.runCVAlgorithm(pressedPoints, threshold);
-                const approx = openCVWrapper.contours.approxPoly(
+                this.latestPoints = await this.runCVAlgorithm(pressedPoints,
+                    toolsBlockerState.algorithmsLocked ? 0 : threshold);
+                const [x, y] = this.latestPoints.slice(-2);
+                if (toolsBlockerState.algorithmsLocked) {
+                    // disable approximation for lastest two points to disable fickering
+                    this.latestPoints.splice(this.latestPoints.length - 2, 2);
+                }
```

Comment:
Only if the algorithm is locked?

Reply:
Yes. I dont think we need to remove one point when we are using normal scissors mode. It will worsen approximation effect